File: cpp/srikanth_mthree.cpp
#include<bits/stdc++.h>
using namespace std;


int diagonalSum(int *ar, int m, int n){

    // cout << "inside function" << endl;
    // cout << m << n << endl;
    int mx = INT_MIN;
    int arr[m][n];
    
    for(int i=0; i<m; i++){
        for(int j=0; j<n; j++){
            // cout << i << " " << j << " ";
            arr[i][j] =  *(ar + i*m + j);
            // cout << arr[i][j];
        }
    }

    for(int k=0; k<m; k++){
        int psm = 0;
        int ssm = 0;
        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                if(i == j+k){           // upper triangular matrix
                    psm+=arr[i][j];
                }

                if(i+j == m-1-k){       // lower triangular matrix
                    ssm+=arr[i][j];
                }
            
            }
            
        }
        // cout << psm << " " << ssm;
        // cout << endl;
        int mxx = (psm<ssm?ssm:psm);
        mx = mx<mxx? mxx: mx;

    }


    return mx;
}

int main(){

    int n,m;
    
    cin >> m >> n ;

    int ar[m][n];

    for(int i=0; i<m; i++){
        for(int j=0; j<n; j++){
            cin >> ar[i][j];
        }
    }

    cout << diagonalSum(ar[0], m, n);

    return 0;
}

File: py/srikanth_mthree.py

def highestSum(m,n,arr):

    mx=-999

    for k in range(0,m):
        psm=0
        ssm=0
        for i in range(0,m):
            for j in range(0,n):
                if i == j+k:
                    psm+=arr[i][j]
                if i+j == m-k-1:
                    ssm+=arr[i][j]

        mx1 = max(psm, ssm)
        mx = max(mx, mx1)


    return mx;


def main():
    m,n = [int(x) for x in input().split()]
    arr=[]
    for i in range(m):
        a = list(map(int, input().split()))
        arr.append(a)
    
    print(highestSum(m,n,arr))


main()
